src/dotnet/ProjectInstanceExtensions.cs

```diff
@@ -22,29 +22,12 @@ public static string GetProjectId(this ProjectInstance projectInstance)
 
         public static string GetProjectTypeGuid(this ProjectInstance projectInstance)
         {
-            string projectTypeGuid = null;
-
-            var projectTypeGuidProperty = projectInstance.GetPropertyValue("ProjectTypeGuid");
-            if (!string.IsNullOrEmpty(projectTypeGuidProperty))
-            {
-                projectTypeGuid = projectTypeGuidProperty.Split(';').Last();
-            }
-            else
+            var projectTypeGuids = projectInstance.GetPropertyValue("ProjectTypeGuids");
```

Comment:
I believe that VS actually scans for the guid using strstr. E.g. you can comment it out and it still has an impact! I believe the rationale is that a project system needn't use msbuild or even xml for its project file format. It might be worth hunting down what VS does and replicating it.

Reply:
Looks okay apart from this case; ProjectTypeGuids is only ever read from the project file itself. DefaultProjectTypeGuid should be read using evaluation, however.

Reply:
Is there a way to limit the evaluation scope or are you suggesting we read the project file XML and search for `ProjectTypeGuids`?

Reply:
Yes, we walk the construction model in other cases similar to this.

Reply:
Gotcha, I'll see if I can do it the way VS does.

Reply:
VS itself actually does a text scan like @nguerrera was suggesting - but walking the construction model will be good enough.

Reply:
@davkean I've pushed up reading the project type guid from the `ProjectRootElement`.  Would you remind reviewing the commit again?  Thanks!